phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataRegionObserver.java

```diff
@@ -202,9 +212,11 @@ public Void run() throws Exception {
                 }
             }
         };
-        Thread t = new Thread(r);
-        t.setDaemon(true);
-        t.start();
+
+        if (env.getConfiguration()
+                .getBoolean(STATS_COLLECTION_ENABLED, DEFAULT_STATS_COLLECTION_ENABLED)) {
+            truncateTaskExectuor.schedule(r, statsTruncateTaskDelay, TimeUnit.MILLISECONDS);
+        }
```

Comment:
Add an else clause and then you can add a log there "Stats collection is disabled"